packages/dynamic-import-vars/src/dynamic-import-to-glob.js

```diff
@@ -98,9 +114,29 @@ export function dynamicImportToGlob(node, sourceString) {
   }
 
   if (!glob.startsWith('./') && !glob.startsWith('../')) {
-    throw new VariableDynamicImportError(
-      `invalid import "${sourceString}". Variable bare imports are not supported, imports must start with ./ in the static part of the import. ${example}`
-    );
+    const match = glob.match(npmPackageNameAndGlobRegex);
+
+    if (match) {
+      const [, packageName, , packageGlob] = match;
+      const resolvedPackageJson = await r.resolve(`${packageName}/package.json`, id);
+
+      if (!resolvedPackageJson) {
+        throw new VariableDynamicImportError(
+          `invalid import "${sourceString}". Could not resolve module "${packageName}".`
+        );
+      }
+
+      glob = `${path.relative(
+        path.dirname(id),
+        path.dirname(resolvedPackageJson.id)
+      )}/${packageGlob}`;
```

Comment:
What makes me a little uneasy is that while this supports package-relative imports, it completely ignores the `exports` property of `package.json` that may completely turn things upside down. This is in contrast the the `node-resolve` plugin that *does* respect that property. Not sure if you can even import `package.json` if the property is set without including it, but I did not check. In any case, I am not sure how to improve here.